ironfish/src/mining/pool.ts

```diff
@@ -390,4 +406,26 @@ export class MiningPool {
       decimalPrecision
     )
   }
+
+  async notifyStatus(): Promise<void> {
+    const status = await this.getStatus()
+    this.logger.debug(`Mining pool status: ${JSON.stringify(status)}`)
+    this.webhooks.map((w) => w.poolStatus(status))
+  }
+
+  async getStatus(): Promise<MiningPoolStatus> {
+    return {
+      name: this.name,
+      hashRate: await this.estimateHashRate(),
+      miners: this.stratum.clients.size,
+      sharesPending: await this.shares.sharesPendingPayout(),
+    }
+  }
+}
+
+export type MiningPoolStatus = {
+  name: string
+  hashRate: number
+  miners: number
+  sharesPending: number
 }
```

Comment:
I usually inline these, but this is fine too as long as we don't have too many of these. Their only purpose is to match sure the types match on the other side but typescript will do that for us so it tends to just gunk up files when there are too many of them.

Reply:
I'm going to leave this here for now -- it will probably make its way into a message schema in followup changes that add a polling for status